elivepatch_server/resources/dispatcher.py

```diff
@@ -159,10 +159,42 @@ def post(self):
         if not kernel_sources_status:
             return make_response(jsonify({'message': 'gentoo-sources not available'}), 403)
         lpatch.build_livepatch('vmlinux', jobs=self.cmdline_args.jobs)
+        livepatch_full_path = os.path.join(uuid_dir, 'kpatch-main.ko')
+        livepatch_saving_folder = os.path.join('/tmp/', 'livepatch', args['UUID'])
+        livepatch_saving_file = os.path.join(livepatch_saving_folder, 'livepatch.ko')
+        if os.path.exists(livepatch_saving_folder):
+            print('saving livepatch to: '+ str(livepatch_saving_file))
+            try:
+                shutil.move(livepatch_full_path, livepatch_saving_file)
+            except:
+                print('live patch not generated')
+                print('check build.log at:' + str(livepatch_saving_folder + '/build.log'))
+                try:
+                    shutil.move(uuid_dir + '/kpatch/build.log', livepatch_saving_folder + '/build.log')
+                except:
+                    print('no build.log generated')
+        else:
+            print('creating: "' + str(livepatch_saving_folder) + '"')
+            os.makedirs(livepatch_saving_folder)
+            print('saving livepatch to: '+ str(livepatch_saving_file))
+            try:
+                shutil.move(livepatch_full_path, livepatch_saving_file)
+            except:
+                print('live patch not generated')
+                print('check build.log at:' + str(livepatch_saving_folder + '/build.log'))
+                try:
+                    shutil.move(uuid_dir + '/kpatch/build.log', livepatch_saving_folder + '/build.log')
+                except:
+                    print('no build.log generated')
 
         pack = {
            'id': packs['id'] + 1,
             'KernelVersion': None,
             'UUID' : args['UUID']
         }
         return {'get_config': marshal(pack, pack_fields)}, 201
+
+    def __del__(self):
```

Comment:
Did you test this under heavy load? I have been rethinking this solution. Although it makes sense to handle this kind of stuff inside a destructor from an object oriented perspective, I have my suspicions about the behavior of the garbage collector. I am also not sure if it's technically sound to use shutil [due to](https://docs.python.org/3/reference/datamodel.html#object.__del__):
```
when __del__() is invoked in response to a module being deleted
(e.g., when execution of the program is done), other globals referenced
by the __del__() method may already have been deleted or in the
process of being torn down (e.g. the import machinery shutting down).
For this reason, __del__() methods should do the absolute minimum
needed to maintain external invariants.
```

I would think that a garbage collector does not have to run immediately when the reference count of the object reaches zero since it may be beneficial to opportunistically delay the garbage collection and save CPU cycles. In this scenario, we may be filling the server's file system because the garbage collector is too lazy.

Reply:
have you any other solution in mind ?